Travis CI builds fail when backend is QtAgg

The ubuntu version of the Travis CI builds has changed to 16.04 (xenial) and the following setup line fails since xvfb does not exist:
```
$ if [ $BACKEND == "qtagg" ]; then export DISPLAY=:99.0; sh -e /etc/init.d/xvfb start; sleep 3; fi
sh: 0: Can't open /etc/init.d/xvfb
```
The failure is at the first test (this is the python2.7 test output but is similar to python3.7):
```
============================= test session starts ==============================
platform linux2 -- Python 2.7.16, pytest-4.6.2, py-1.8.0, pluggy-0.12.0
rootdir: /home/travis/build/mwaskom/seaborn
plugins: cov-2.7.1
collected 564 items                                                            
seaborn/axisgrid.py Makefile:10: recipe for target 'coverage' failed
make: *** [coverage] Error 1
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -25,7 +25,7 @@ before_install:
   - export PATH="$HOME/miniconda/bin:$PATH"
   - hash -r
   - conda config --set always_yes yes --set changeps1 no
-  - conda update -q conda openssl  # Update openssl to fix python 2.7.9 issue
+  - conda update -q conda
   - conda info -a
 
 
@@ -43,6 +43,10 @@ before_script:
     export DISPLAY=:99.0;
     sh -e /etc/init.d/xvfb start;
     sleep 3;
+    fi
+    # https://www.python.org/dev/peps/pep-0493/
+  - if [ $PYTHON == "2.7" ]; then
+    export PYTHONHTTPSVERIFY=0;
     fi
 
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -25,7 +25,7 @@ before_install:
   - export PATH="$HOME/miniconda/bin:$PATH"
   - hash -r
   - conda config --set always_yes yes --set changeps1 no
-  - conda update -q conda
+  - conda update -q conda openssl  # Update openssl to fix python 2.7.9 issue
   - conda info -a
 
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,6 +1,11 @@
 language: python
 
 
+dist: xenial
+services:
+  - xvfb
+
+
 env:
   - PYTHON=2.7 DEPS=latest BACKEND=agg DOCTESTS=true
   - PYTHON=2.7 DEPS=pinned BACKEND=agg DOCTESTS=false

--- a/seaborn/tests/test_axisgrid.py
+++ b/seaborn/tests/test_axisgrid.py
@@ -436,6 +436,7 @@ class TestFacetGrid(object):
         nt.assert_equal(g.axes[0, 1].get_title(), "b == n")
         nt.assert_equal(g.axes[1, 0].get_title(), "")
 
+    @pytest.mark.xfail(mpl.__version__ == "3.1.0", reason="matplotlib bug")
     def test_set_ticklabels(self):
 
         g = ag.FacetGrid(self.df, row="a", col="b")

